latest_builds/matrix.yml

```
@@ -49,6 +49,21 @@ testing-ansible-12:
       dists:
         - noble
         - plucky
+testing-ansible-13:
+  github_branch: ansible-13
+  packages:
+    - name: ansible-core
+      version_specifier: "~=2.20.0a"
```

ansible-core has beta releases, so shouldn't it be
```suggestion
      version_specifier: "~=2.20.0b"
```
?

---

Hey @felixfontein, thanks for taking a look!

That's a great question. Since I'm not 100% sure which part you're asking about, I'll try to cover all the bases.

First, just in case the question is about how the specifier itself works: `~=2.20.0a` does correctly catch betas. It's being used to create a [SpecifierSet](https://packaging.pypa.io/en/stable/specifiers.html) [here](https://github.com/ansible-community/ppa/blob/main/latest_builds/latest_builds.py#L112) and translates to `>=2.20.0a, <2.21.0`, which includes alphas, betas, release candidates, and stable releases.

However, your question could also be about the *intent* behind using `a`. The `a` is deliberate for our "testing" PPAs builds to create a catch-all for any pre-release (alphas, betas, etc.). The builds without an `a` in the specifier are for stable releases only.

Your question makes me realize this logic isn't very obvious. What do you think about me adding a quick comment to explain this pattern? I'm also thinking of renaming the variable to version_specifier_set for more clarity. Let me know what you think. Thanks again!

---

I think adding a comment that this covers every version in range `>=2.20.0a, <2.21.0` would help. Without knowing the exact format, I would guess it means "any 2.20.0 alpha version".

(Or a more generic comment at the top of the file, so you don't have to copy around the comments and also modify them...)

---

Not to put too fine a point on this, but would it be possible to have `version_specifier: ">=2.20.0a, <2.21.0"` here? Then nobody would need a comment to understand what `~=2.20.0a` means. (Actually, wouldn't `~=2.20.0a` be compatible with 2.21.0 and later under semver rules? Which ansible-core unfortunately doesn't follow... I'm not sure there, though) This would also avoid the situation that the comment has to be updated constantly, and also that it's possibly wrong or at least not consistent with the actual version_specifier.

Explicit is better than implicit. Just a thought :smirk:

---

Pushed some updates... Let me know what you think @felixfontein & @mariolenz

---

Yes, that what I meant. What do you think about this explicit version specifier @felixfontein?

---

Definitely more clear for persons like me which forgot about the exact specifier set syntax ;-)